Find a functionally equivalent code:com.android.internal.os.ZygoteConnection.Arguments.parseArgs:COMMENT
Method Modifier: private     internal    
Comment:/**
 * Parses the commandline arguments intended for the Zygote spawner
 * (such as "--setuid=" and "--setgid=") and creates an array
 * containing the remaining args.
 *
 * Per security review bug #1112214, duplicate args are disallowed in
 * critical cases to make injection harder.
 */

@@ -106,111 +106,117 @@
             mountExternal = Zygote.MOUNT_EXTERNAL_MULTIUSER;
         } else if (arg.equals("--mount-external-multiuser-all")) {
             mountExternal = Zygote.MOUNT_EXTERNAL_MULTIUSER_ALL;
+        } else if (arg.equals("--query-abi-list")) {
+            abiListQuery = true;
+        } else if (arg.startsWith("--instruction-set=")) {
+            instructionSet = arg.substring(arg.indexOf('=') + 1);
+        } else if (arg.startsWith("--app-data-dir=")) {
+            appDataDir = arg.substring(arg.indexOf('=') + 1);
         } else {
             break;
         }

